sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -94,4 +108,21 @@ object InternalRow {
 
   /** Returns an empty [[InternalRow]]. */
   val empty = apply()
+
+  /**
+   * Copies the given value if it's string/struct/array/map type.
+   */
+  def copyValue(value: Any): Any = {
+    if (value.isInstanceOf[UTF8String]) {
```

Comment:
??? - pattern matching should yield the same performance as `if (value.instanceof[...]) ... else if (value.instanceof[...]) ...`